### PR TITLE
added support for redis auth

### PR DESCRIPTION
**Pull request by el4v**
**IMPORTANT: Please do not create a Pull Request without creating an issue first!**

_(Any change needs to be discussed before proceeding. Failure to do so may result in the rejection of the pull request)._

### Summary
Added support for REDIS_AUTH

### Type of Change
- Added support for REDIS_AUTH

### How has this been tested?
Proof:

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly.
- [ ] I have written new tests for my changes.
  - [x] My changes successfully ran and pass tests locally.

### CODE REVIEW

File: pkg/service/configmap_env_builder.go [hunk 1]
```diff
@@ -58,6 +58,7 @@ func (n *EnvBuilder) buildRedisEnv() (map[string]string, error) {
 	data["USE_STATSD"] = "false"
 	data["REDIS_TYPE"] = n.RateLimitService.Spec.Backend.Redis.Type
 	data["REDIS_URL"] = n.RateLimitService.Spec.Backend.Redis.URL
+	data["REDIS_AUTH"] = n.RateLimitService.Spec.Backend.Redis.Auth
```
Comment (zufardhiyaulhaq) [inline pkg/service/configmap_env_builder.go:61]: not all Redis implementation is using auth, can we check if `n.RateLimitService.Spec.Backend.Redis.Auth` and assign to data map if it's exist?

**Comment by zufardhiyaulhaq** [comment]
@el4v thanks for contributing. can you make the DCO pass the test first?

**Comment by el4v** [comment]
> @el4v thanks for contributing. can you make the DCO pass the test first?

done

**Comment by zufardhiyaulhaq** [comment]
@el4v also DCO is still red
gambar

**Pull request closed without merging**